app/javascript/components/student/Editor.tsx

```diff
@@ -58,6 +59,7 @@ export enum TestStatus {
 type SubmissionLinks = {
```

Comment:
At some point I recommend to switch to

```typescript
{
  cancel: { href: string },
  submit: { href: string },
  testRun: { href: string }
}
```

It allows for metadata and parameters to be part of the URL value, and doesn't require adding weird types later to do this.

Reply:
Cool. I'll put this in a separate issue.